master/internal/rm/agentrm/agent_resource_manager.go

```diff
@@ -419,9 +423,12 @@ func (a *ResourceManager) RecoverJobPosition(msg sproto.RecoverJobPosition) {
 
 // Release implements rm.ResourceManager.
 func (a *ResourceManager) Release(msg sproto.ResourcesReleased) {
-	for _, pool := range a.pools {
-		pool.ResourcesReleased(msg)
+	pool, err := a.poolByName(msg.ResourcePool)
+	if err != nil {
+		a.syslog.WithError(err).Warnf("release found no resource pool with name %s",
```

Comment:
return otherwise the next line will nil deref (with resp, err := f() usually err != nil => resp == nil, you can usually think of a (T, error) signature as a result<T> in other languages (rust))

Reply:
Ooh gotchya!